packages/dds/tree/src/shared-tree/schematizingTreeView.ts

```diff
@@ -301,73 +300,61 @@ export class SchematizingSimpleTreeView<
 	 * If implicit initialization (or some other edit) is desired, it should be done outside of this method.
 	 */
 	private update(): void {
-		this.disposeView();
+		this.disposeFlexView();
 
 		const compatibility = this.viewSchema.checkCompatibility(this.checkout.storedSchema);
 
-		let lastRoot =
-			this.compatibility.canView && this.view !== undefined ? this.root : undefined;
+		let lastRoot: ReadableField<TRootSchema> | undefined;
```

Comment:
The call to disposeView above sets `this.view === undefined` so only the undefined case here was used.

Reply:
I have now refactored the code to make this more clear (instead of just blindly simplifying the old code). That also found a bug I fixed which now has a regression test and changeset for it included. Should be more clear now.